web/src/lib/components/shared-components/search-bar/search-filter-box.svelte

```diff
@@ -2,169 +2,373 @@
-          >
-            <input
-              bind:group={mediaType}
-              value={MediaType.Video}
-              type="radio"
-              name="radio-type"
-              id="type-video"
-            />Video</label
-          >
-        </div>
-      </div>
+    mediaType: MediaType;
+  };
 
-      <!-- DISPLAY OPTIONS -->
-      <div id="display-options-selection">
-        <p class="immich-form-label">DISPLAY OPTIONS</p>
+  let suggestions: SearchSuggestion = {
+    people: [],
+    country: [],
+    state: [],
+    city: [],
+    cameraMake: [],
+    cameraModel: [],
+  };
 
-        <div class="flex gap-5 mt-3">
-          <label class="flex items-center mb-2">
-            <input type="checkbox" class="form-checkbox h-5 w-5 color" bind:checked={notInAlbum} />
-            <span class="ml-2 text-sm text-black dark:text-white pt-1">Not in any album</span>
-          </label>
+  let filter: SearchFilter = {
+    context: undefined,
+    people: [],
+    location: {
+      country: undefined,
+      state: undefined,
+      city: undefined,
+    },
+    camera: {
+      make: undefined,
+      model: undefined,
+    },
+    dateRange: {
+      startDate: undefined,
+      endDate: undefined,
+    },
+    inArchive: undefined,
+    inFavorite: undefined,
+    notInAlbum: undefined,
+    mediaType: MediaType.All,
+  };
 
-          <label class="flex items-center mb-2">
-            <input type="checkbox" class="form-checkbox h-5 w-5 color" bind:checked={inArchive} />
-            <span class="ml-2 text-sm text-black dark:text-white pt-1">Archive</span>
-          </label>
+  let showAllPeople = false;
+  $: subPeopleList = showAllPeople ? suggestions.people : suggestions.people.slice(0, 11);
 
-          <label class="flex items-center mb-2">
-            <input type="checkbox" class="form-checkbox h-5 w-5 color" bind:checked={inFavorite} />
-            <span class="ml-2 text-sm text-black dark:text-white pt-1">Favorite</span>
-          </label>
-        </div>
-      </div>
-    </div>
+  onMount(() => {
+    getPeople();
+  });
+
+  const sortPeople = () => {
+    suggestions.people.sort((a, _) => {
+      if (filter.people.some((p) => p.id === a.id)) {
+        return -1;
+      }
+      return 1;
+    });
+  };
+
+  const getPeople = async () => {
+    try {
+      const { data } = await api.personApi.getAllPeople({ withHidden: false });
+      suggestions.people = data.people;
+    } catch (error) {
+      handleError(error, 'Failed to get people');
+    }
+  };
+
+  const handlePeopleSelection = (id: string) => {
+    if (filter.people.some((p) => p.id === id)) {
+      filter.people = filter.people.filter((p) => p.id !== id);
+      sortPeople();
+      return;
+    }
+
+    const person = suggestions.people.find((p) => p.id === id);
+    if (person) {
+      filter.people = [...filter.people, person];
+      sortPeople();
+    }
+  };
```

Comment:
This code is difficult to understand. The first part removes the id from the filtered people if it exists and the second part adds it if it doesn't. So it's like a toggle? And `filter.people` is used in `sortPeople` to show selected people first and for styling. Is there a significance to finding the person entry as a whole and adding it to `filter.people` instead of using the id?

I think it'd be a bit clearer (and faster) if you made `filter.people` a set.  Renaming the `filter` object to `selected` would also make the intent more explicit. And if you changed `sortPeople` to `showSelectedPeopleFirst`, it'd convey what it's sorting by.
```typescript
  const handlePeopleSelection = (id: string) => {
    if (selected.people.has(id)) {
      selected.people.delete(id);
    } else {
      selected.people.add(id);
    }
    showSelectedPeopleFirst();
  };
```

Reply:
The value in the array is `PersonReponseDto`, it is an object so it doesn't really work with `Set`. We will need the object to render the name and thumbnail in the people list

Reply:
`filter.people` as it is only seems to be used to check if a person ID exists in it. I don't think it needs to have the person object itself for this. It can just be a set of IDs. `suggestions.people` is what's used to get the name etc. in the current code.